packages/host/app/resources/card-type.ts

```diff
@@ -35,12 +35,21 @@ interface Args {
     loader: Loader;
   };
 }
+
+export type CodeRefType = CodeRef & {
+  displayName: string;
+};
+
```

Comment:
I feel something weird and unusual. 
- code-ref seems to treat field as children of cards
- card-type sorta treats fields the same as cards in that a `Type`

Which leads me to this question. Cudn you get `displayName` anyway without making the changes to the card-type file

Reply:
Ahh I think I can see why actually. It has to do with the way the `card-schema-editor` works and how it uses the `fields` property

Reply:
Yes, I could retrieve `displayName` without making changes to this file. I considered two solutions: the first involves updating the card-type file, which is what this PR does, and the second involves updating the `fieldDisplayName` function in the card-schema-editor file.

I believed it would be better if the `carTypeResource` could provide all the data that clients need. This way, we can keep the logic regarding 'field refers to itself' in this file. Before this PR, we returned the card of Type's field as `CodeRef`, so I thought it was fine to "add" `displayName` to the `CodeRef`.

Reply:
I think this is fine. A `CodeRef` is a way to point to where some executable code lives. To create a type that can point to a particular executable code and give it a "name" as well seems like a reasonable modification